docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,316 @@
+
+{{< /hint >}}
+
+#### 6.1 Run grept
+
+[Grept](https://github.com/Azure/grept) is a linting tool for repositories, ensures predefined standards, maintains codebase consistency, and quality.
+It's using the grept configuration files from the [Azure-Verified-Modules-Grept](https://github.com/Azure/Azure-Verified-Modules-Grept) repository.
+
+You can see [here](https://github.com/Azure/Azure-Verified-Modules-Grept/blob/main/terraform/synced_files.grept.hcl) which files are synced from the [`terraform-azurerm-avm-template`](https://github.com/Azure/terraform-azurerm-avm-template) repository.
+
+1. Set environment variables
+
+```bash
+# Linux/MacOS
+export GITHUB_REPOSITORY_OWNER=Azure
+export GITHUB_REPOSITORY=Azure/terraform-azurerm-avm-res-<RP>-<modulename>"
+
+# Windows
+
+$env:GITHUB_REPOSITORY_OWNER="Azure"
+$env:GITHUB_REPOSITORY="Azure/terraform-azurerm-avm-res-<RP>-<modulename>"
+```
+
+1. Run grept
+
+```bash
+grept apply -a git::https://github.com/Azure/Azure-Verified-Modules-Grept.git//terraform
+```
+
+#### 6.2. Check/Generate module documentation
+
+```bash
+./avm docscheck # comparing generated README.md with the one in the repo
+./avm docs # generating module documentation like README.md including examples
+```
+
+#### 6.3 Format Terraform code
+
+```bash
+./avm fmt
+```
+
+#### 6.4 Run e2e tests
+
+<br>
+
+---
+
+<br>
+
+### 7. Create a pull request to the upstream repository
+
+Once you are satisfied with your contribution and validated it, open a PR from your forked repository to the original Terraform Module repository. Make sure you:
+
+1. Include/Add [`@Azure/avm-core-team-technical`](https://github.com/orgs/Azure/teams/avm-core-team-technical/members) as a reviewer.
+2. Make sure all Pull Requst Checks (e2e tests with all examples, linting and version-check) are passing.
+
+<br>
+
+---
+
+<br>
+
+### Common mistakes to avoid and recommendations to follow
+
+<!--
+TODO:
+- might be worth adding `terraform.tfvars` to `.gitignore` ? -->
+
+- If you contribute to a new module then search and update `TODOs` (which are coming with the [terraform-azurerm-avm-template](https://github.com/Azure/terraform-azurerm-avm-template)) within the code and remove the `TODO` comments once complete
+- `terraform.lock.hcl` shouldn't be in the repository as per the `.gitignore` file
+- Update the `support.md` file
+- Consider following specs [TFNFR31](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr31---category-code-style---localstf-must-contain-only-one-locals-block) for the `local.tf` file
```

Comment:
This is incorrect, I will make a PR to change - see #522

Reply:
Removed.

Reply:
Adjusted.